utils/hooks.ts

```diff
@@ -176,10 +179,30 @@ function useUnsavedChangesWarning({
   }, [isDirty, router.events, isSubmitted, warningMessage]);
 }
 
+function useRedirectAuthenticatedUsers() {
+  const router = useRouter();
+  const { authState } = useAuthContext();
+
+  const authenticated = authState.authenticated;
+  const redirectFrom = stringOrFirstString(router.query.from) ?? dashboardRoute;
+  const redirectTo =
+    redirectFrom === "/" || redirectFrom === "%2F"
+      ? dashboardRoute
+      : redirectFrom;
+
+  useEffect(() => {
+    if (authenticated) {
+      router.push(redirectTo);
+    }
+  }, [authenticated, redirectTo]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Rather than omitting `router` in the dependency array, it seems like this alternative suggested workaround of using `Router.push` (default export from `next/router`) is safer - https://github.com/vercel/next.js/issues/18127#issuecomment-988959843 - as it can be easy to modify this and forget to update the dep arrays in the future.

Using `Router` should be ok in this case as we are doing a blanket redirect anyway rather than requiring any route specific state